pgxn/neon/control_plane_connector.c

```diff
@@ -433,32 +433,30 @@ MergeTable()
 											   entry->name,
 											   HASH_ENTER,
 											   NULL);
+			RoleEntry * old;
+			bool found_old = false;
 
 			to_write->type = entry->type;
-			if (entry->password)
-				to_write->password = entry->password;
+			to_write->password = entry->password;
 			strlcpy(to_write->old_name, entry->old_name, NAMEDATALEN);
-			if (entry->old_name[0] != '\0')
-			{
-				bool		found_old = false;
-				RoleEntry  *old = hash_search(
-											  CurrentDdlTable->role_table,
-											  entry->old_name,
-											  HASH_FIND,
-											  &found_old);
-
-				if (found_old)
-				{
-					if (old->old_name[0] != '\0')
-						strlcpy(to_write->old_name, old->old_name, NAMEDATALEN);
-					else
-						strlcpy(to_write->old_name, entry->old_name, NAMEDATALEN);
-					hash_search(CurrentDdlTable->role_table,
-								entry->old_name,
-								HASH_REMOVE,
-								NULL);
-				}
-			}
+			if (entry->old_name[0] == '\0')
+				continue;
+
+			old = hash_search(
+							  CurrentDdlTable->role_table,
+							  entry->old_name,
+							  HASH_FIND,
+							  &found_old);
+			if (!found_old)
+				continue;
+			if (old->old_name[0] != '\0')
```

Comment:
How `old_name` can be empty string if we locate it using has lockup with non-empty key (see `entry->old_name[0] == '\0'` check above)?

Reply:
I'm not the original code author but yes, this is suspicious, fixed